### ISSUE TITLE
Rect parameter in training function is not working

### ISSUE THREAD

**Issue by lolyafedor** [ultralytics/ultralytics#8198]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

When using rect = True, parameter in function train() it's not padding data or making it rectangular.
Here's a pic of the training batch with ultralytics 8.1.13
![image](https://github.com/ultralytics/ultralytics/assets/114881524/9347ee15-9051-4fbc-ad14-41e41b0e1649)
Here's a pic of the training batch with ultralytics 8.0.218
![image](https://github.com/ultralytics/ultralytics/assets/114881524/91215114-16d3-4d08-a93b-3d235a22dcd7)
Here's an example of the original image
![image](https://github.com/ultralytics/ultralytics/assets/114881524/aadba77a-a794-4200-97b0-cd0d4242e692)

### Environment

Ultralytics YOLOv8.1.13 🚀 Python-3.11.5 torch-2.2.0+cu121 CPU (AMD EPYC 7302 16-Core Processor)
Setup complete ✅ (32 CPUs, 251.6 GB RAM, 731.6/915.3 GB disk)

### Minimal Reproducible Example

results = model.train(data = "", project = "", device = 'cpu', rect=True )

### Additional

All this code is used in the jupyter notebooks

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@lolyafedor hello! Thanks for bringing this to our attention. It seems like the `rect` parameter isn't functioning as expected in your training setup. The `rect` parameter is designed to enable rectangular training, which should adjust the padding of your images to maintain their aspect ratio.

Could you please try setting the `rect` parameter directly in your dataset YAML file instead of passing it as an argument in the `train()` function? Here's a quick example of how you can set it in your YAML:

```yaml
# dataset.yaml
train: path/to/train/images
val: path/to/val/images
rect: True
...
```

Then, you can initiate training without specifying `rect` in the `train()` function:

```python
results = model.train(data="dataset.yaml", project="", device='cpu')
```

If the issue persists after this change, it would be helpful if you could open an issue with the details on our GitHub repository, and we'll investigate further. Thanks for your contribution to improving YOLOv8! 🌟

**Comment by lolyafedor** [comment]
I've tried to pass it a .yaml file , but it gives an error that it expects a directory with train folder
<<RuntimeError: Dataset 'dataset.yaml' error ❌ [Errno 2] No such file or directory: '/mnt/1T_SSD/MachineLearningProjects/students2/valentin/DGM_Documents/modeling/datasets/dataset.yaml/train'>>
Thats my yaml file
![image](https://github.com/ultralytics/ultralytics/assets/114881524/f3462de0-7c60-4ee7-b80d-34b7722aeedc)
Thats my code
<<results = model.train(data = '/mnt/1T_SSD2/interns/intern_9/ClassificationModel/dataset.yaml', project = os.path.join('runs',original_dataset_name), device = 'cpu', exist_ok = False,
                      epochs = 25, hsv_h = 0.25,hsv_s = 1,hsv_v = 0.8,mixup = 0.5)>>
I've seen your reply that we cant use a yaml file when we're classifying smth so, is it still correct?
https://github.com/ultralytics/ultralytics/issues/4078#issuecomment-1660831150

**Comment by glenn-jocher** [comment]
@lolyafedor thank you for reporting this issue with the `rect` parameter during training. It seems like there might be a regression or a bug that's causing the rectangular training (`rect=True`) to not work as expected in the latest version you mentioned (8.1.13).

To address this, we'll need to investigate the changes between the versions 8.0.218 and 8.1.13 to identify any modifications that could have affected the rectangular training functionality. In the meantime, you can use the following steps as a workaround:

1. **Use the Last Known Good Version**: If the rectangular training was working as expected in version 8.0.218, you might consider downgrading to that version temporarily until the issue is resolved in the newer version.

2. **Check for Updates**: Make sure you have the latest version of the YOLOv8 repository, as the issue might have been fixed in a newer release after 8.1.13.

3. **Submit a Bug Report**: Since you've checked the existing issues and didn't find a similar report, please submit a detailed bug report on the Ultralytics YOLOv8 GitHub repository. Include the information you've provided here, along with the images and the minimal reproducible example.

4. **Submit a PR**: If you're familiar with the codebase and you're willing to help by submitting a PR, you could try to identify the commit that introduced the bug by comparing the relevant sections of the code between the two versions. If you find the cause, you can make the necessary changes and submit a PR for the maintainers to review.

Please make sure to follow the contribution guidelines provided by the Ultralytics team when submitting your PR. Your contribution would be greatly appreciated by the community.

If you need further assistance with creating the PR or if you have additional information that could help resolve this issue, please let us know.
